web/src/common/Intersection.ts

```diff
@@ -0,0 +1,42 @@
+import type { Feature, Point } from "geojson";
+
+export type IntersectionId = number;
+
+export type DiagonalFilter = {
+  angle: number;
+};
+
+export type IntersectionProperties = {
+  kind: "editable_intersection";
+  intersection_id: number;
+  filter?: DiagonalFilter;
+};
+export type IntersectionFeature = Feature<Point, IntersectionProperties>;
+
+export class Intersection {
```

Comment:
I've gotten into the habitof not making classes in all my Svelte projects, partly because Svelte 4 has some oddness with reactivity outside of `.svelte` files. Svelte 5 fixes all of that, but I haven't thought through how things might be written differently now.

This is fine here and nice to see; I'll think about how other parts of the code could be more organized.